internal/functions/metricsapi/prepare-new-version.go

```diff
@@ -0,0 +1,228 @@
+		httputils.SendErrorResponse(w, r, err)
+		return
+	}
+
+	activationsYesterday, activationsTotal, err := getActivationCounters(ctx, now)
+	if err != nil {
+		logger.Errorf("Error while fetching data: %v", err)
+		httputils.SendErrorResponse(w, r, err)
+		return
+	}
+
+	publishersYesterday, publishersTotal, err := getPublishersCount(ctx)
+	if err != nil {
+		logger.Errorf("Error while fetching data: %v", err)
+		httputils.SendErrorResponse(w, r, err)
+		return
+	}
+
+	var date = now.Add(time.Hour * -24).Format("20060102")
+
+	var data = MetricsData{
+		Modified:               now.Unix(),
+		Date:                   date,
+		ActivationsYesterday:   activationsYesterday,
+		ActivationsTotal:       activationsTotal,
+		KeyPublishersYesterday: publishersYesterday,
+		KeyPublishersTotal:     publishersTotal,
+		NotificationsYesterday: notificationsYesterday,
+		NotificationsTotal:     notificationsTotal,
+	}
+
+	firestoreClient := store.Client{}
+
+	_, err = firestoreClient.Doc(constants.CollectionMetrics, date).Set(ctx, &data)
+	if err != nil {
+		logger.Errorf("Error while saving data: %v", err)
+		httputils.SendErrorResponse(w, r, err)
+		return
+	}
+
+	logger.Infof("Successfully written metrics data to firestore: %+v", data)
+
+	httputils.SendResponse(w, r, struct{ status string }{status: "OK"})
+}
+
+func getNotificationCounters(ctx context.Context, now time.Time) (yesterday int32, total int32, err error) {
+	logger := logging.FromContext(ctx)
+
+	var date = now.Add(time.Hour * -24).Format("20060102")
+
+	// yesterday
+
+	yesterdayData, err := getNotificationCounter(ctx, date)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	logger.Infof("Notifications count for yesterday: %v", yesterdayData.NotificationsCount)
+
+	// total
+
+	totalData, err := getNotificationCounter(ctx, "total")
+	if err != nil {
+		return 0, 0, err
+	}
+
+	logger.Infof("Notifications total count: %v", totalData.NotificationsCount)
+
+	return int32(yesterdayData.NotificationsCount), int32(totalData.NotificationsCount), nil
+}
+
+func getNotificationCounter(ctx context.Context, key string) (*structs.NotificationCounter, error) {
+	logger := logging.FromContext(ctx)
+	var storeClient = store.Client{}
+
+	logger.Debugf("Getting notification counter with key %v", key)
+
+	doc := storeClient.Doc(constants.CollectionNotificationCounters, key)
+
+	// TODO handle not found
+
+	rec, err := doc.Get(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Error while querying Firestore: %v", err)
+	}
+
+	var data structs.NotificationCounter
+	err = rec.DataTo(&data)
+	if err != nil {
+		return nil, fmt.Errorf("Error while querying Firestore: %v", err)
+	}
+	return &data, nil
+}
+
+func getActivationCounters(ctx context.Context, now time.Time) (yesterday int32, total int32, err error) {
+	logger := logging.FromContext(ctx)
+
+	var date = now.Add(time.Hour * -24).Format("20060102")
+
+	// yesterday
+
+	yesterdayData, err := getActivationCounter(ctx, date)
+	if err != nil {
+		return 0, 0, err
```

Comment:
would have been much nicer and cleaner if the yesterday and total were returned together as a struct :)